NGitLab.Mock/MergeRequestCollection.cs

```diff
@@ -45,7 +45,7 @@ public override void Add(MergeRequest mergeRequest)
 
             if (mergeRequest.Iid == default)
             {
-                mergeRequest.Iid = GetNewIid();
+                mergeRequest.Iid = Server.GetNewMergeRequestId();
```

Comment:
In a project, `Iid` are incremented values => This logic doesn't consider the project, so it cannot be valid.

As mentioned in https://github.com/ubisoft/NGitLab/issues/168#issuecomment-1024642870, I think we should just change the initial value of `_lastMergeRequestId`. This way `IId` will be incremented values but will always be different from the `Id`.

Changing the initial `Id` should not impact anyone as there is no reason to use the `Id` property in the context of the GitLab API (this is a deprecated info)

Reply:
To be clear, I think we should set a default value for all these fields (could be random or a high value such as 10_000) https://github.com/ubisoft/NGitLab/blob/main/NGitLab.Mock/GitLabServer.cs#L15